core/src/main/java/de/jplag/merging/MatchMerging.java

```diff
@@ -0,0 +1,136 @@
+ * When neighboring matches get merged they become one and the tokens separating them get removed from the submission
+ * clone. MergeBuffer describes how shorter a match can be than the Minimum Token Match. SeperatingThreshold describes
+ * how many tokens can be between two neighboring matches. Both are set in {@link JPlagOptions} as
+ * {@link MergingParameters} and default to 0 (which deactivates merging).
+ */
+public class MatchMerging {
+    private int minimumTokenMatch;
+    private Submission firstSubmission;
+    private Submission secondSubmission;
+    private List<Match> globalMatches;
+    private List<List<Match>> neighbors;
+    private int seperatingThreshold;
+    private JPlagResult result;
+    private List<JPlagComparison> comparisons;
+    private JPlagOptions options;
+
+    /**
+     * Constructor for class MatchMerging
+     * @param result is the initially computed result object
+     * @param options encapsulates the adjustable options
+     */
+    public MatchMerging(JPlagResult result, JPlagOptions options) {
+        this.result = result;
+        this.comparisons = new ArrayList<>(result.getAllComparisons());
+        this.options = options;
+        this.minimumTokenMatch = options.minimumTokenMatch();
+        this.seperatingThreshold = options.mergingParameters().seperatingThreshold();
+    }
+
+    /**
+     * Runs the internal match merging pipeline. It computes neighboring matches, merges them based on
+     * {@link MergingParameters} and removes the merge buffer afterwards.
+     * @return JPlagResult containing the merged matches
+     */
+    public JPlagResult run() {
+        long timeBeforeStartInMillis = System.currentTimeMillis();
+        for (int i = 0; i < comparisons.size(); i++) {
+            firstSubmission = comparisons.get(i).firstSubmission().copy();
+            secondSubmission = comparisons.get(i).secondSubmission().copy();
+            globalMatches = new ArrayList<>(comparisons.get(i).matches());
+            globalMatches.addAll(comparisons.get(i).ignoredMatches());
+            computeNeighbors();
+            mergeNeighbors();
+            removeBuffer();
+            comparisons.set(i, new JPlagComparison(firstSubmission, secondSubmission, globalMatches, new ArrayList<>()));
+
+        }
+        long durationInMillis = System.currentTimeMillis() - timeBeforeStartInMillis;
+        return new JPlagResult(comparisons, result.getSubmissions(), result.getDuration() + durationInMillis, options);
+    }
+
+    private void computeNeighbors() {
+        neighbors = new ArrayList<>();
+        List<Match> sortedByFirst = new ArrayList<>(globalMatches);
+        Collections.sort(sortedByFirst, (m1, m2) -> m1.startOfFirst() - m2.startOfFirst());
+        List<Match> sortedBySecond = new ArrayList<>(globalMatches);
+        Collections.sort(sortedBySecond, (m1, m2) -> m1.startOfSecond() - m2.startOfSecond());
```

Comment:
Regarding naming, I would prefer something like first, second.

Reply:
You are talking about matches m1 and m2?

Reply:
Yes, we try to avoid abbreviations.

Reply:
Does match1 and match2 work for you?
first and second could be ambiguous.